signer/src/transaction_signer.rs

```diff
@@ -256,6 +285,87 @@ where
         Ok(true)
     }
 
+    #[tracing::instrument(skip(self))]
+    async fn handle_wsts_message(
+        &mut self,
+        msg: &message::WstsMessage,
+        bitcoin_chain_tip: &model::BitcoinBlockHash,
+    ) -> Result<(), error::Error> {
+        tracing::info!("handling message");
+        match &msg.inner {
+            wsts::net::Message::DkgBegin(_) => {
+                let state_machine = self.create_new_state_machine()?;
+                self.wsts_state_machines.insert(msg.txid, state_machine);
+                self.relay_message(msg.txid, &msg.inner, bitcoin_chain_tip)
+                    .await?;
+            }
+            wsts::net::Message::DkgPublicShares(_) => {
+                self.relay_message(msg.txid, &msg.inner, bitcoin_chain_tip)
+                    .await?;
+            }
+            wsts::net::Message::DkgPrivateBegin(_) => {
+                self.relay_message(msg.txid, &msg.inner, bitcoin_chain_tip)
+                    .await?;
+            }
+            wsts::net::Message::DkgPrivateShares(_) => {
+                self.relay_message(msg.txid, &msg.inner, bitcoin_chain_tip)
+                    .await?;
+            }
+            wsts::net::Message::DkgEndBegin(_) => {
+                self.relay_message(msg.txid, &msg.inner, bitcoin_chain_tip)
+                    .await?;
+                self.store_dkg_shares(&msg.txid).await?;
+            }
+            wsts::net::Message::NonceRequest(_) => {
+                // TODO(296): Validate that message is the appropriate sighash
+                self.relay_message(msg.txid, &msg.inner, bitcoin_chain_tip)
+                    .await?;
+            }
+            wsts::net::Message::SignatureShareRequest(_) => {
+                // TODO(296): Validate that message is the appropriate sighash
+                self.relay_message(msg.txid, &msg.inner, bitcoin_chain_tip)
+                    .await?;
+            }
+            _ => {
+                tracing::debug!("ignoring message");
+            }
```

Comment:
It seems like we want to handle some of these messages (like `SignatureShareResponse` or something), does that get handled elsewhere?

Reply:
There are three message types not handled here (`DkgEnd`, `NonceResponse` and `SignatureShareResponse`). The `DkgEnd` message should only be relevant for the signer set manager to be able to finish DKG, and the `NonceResponse` and `SignatureShareResponse` should only be relevant for the transaction_coordinator to be able to produce valid signatures. (Assuming the component breakdown from #44)